generator.yaml

```diff
@@ -497,6 +499,78 @@ resources:
         from:
           operation: DescribeMonitoringSchedule
           path: LastMonitoringExecutionSummary
+  ModelPackage:
+    exceptions:
+      errors:
+          404:
+            code: ValidationException
+            message_suffix: does not exist.
+      terminal_codes:
+        - ResourceLimitExceeded
+        - ResourceNotFound
+        - ResourceInUse
+        - OptInRequired
+        - InvalidParameterCombination
+        - InvalidParameterValue
+        - MissingParameter
+        - MissingAction
+        - InvalidQueryParameter
+        - MalformedQueryString
+        - InvalidAction
+        - UnrecognizedClientException    
+        - ConflictException   
+    hooks:
+      delta_pre_compare:
+        code: customSetDefaults(a, b)
+      sdk_delete_pre_build_request:
+        template_path: common/sdk_delete_pre_build_request.go.tpl
+      sdk_delete_post_request:
+        template_path: common/sdk_delete_post_request.go.tpl
+      sdk_delete_post_build_request:
+        template_path: model_package/set_resource_name_as_arn.go.tpl
```

Comment:
modelpackagename can either be the arn in case of Versioned ModelPackage or the modelpackagename in unversioned modelpackage. aws-sdk-go calls the field modelpackagename  and the description says "NAME OR ARN"